### PR TITLE
Validate the Master node CIDR does not change in `AzureConfig` CR.

### PR DESCRIPTION
**Pull request by whites11**
this PR ensures the master CIDR in the AzureConfig CR never changes

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.15
 
 require (
 	github.com/Azure/azure-sdk-for-go v46.4.0+incompatible
-	github.com/Azure/go-autorest/autorest/azure/auth v0.5.5
+	github.com/Azure/go-autorest/autorest/azure/auth v0.5.2
```
Comment (whites11) [inline go.mod:7]: Needed to downgrade or it didn't build

File: pkg/azureupdate/validate_azureconfig.go [hunk 2]
```diff
@@ -64,9 +64,23 @@ func (a *AzureConfigValidator) Validate(ctx context.Context, request *v1beta1.Ad
...
+
 	return nil
 }
 
 func (a *AzureConfigValidator) Log(keyVals ...interface{}) {
 	a.logger.Log(keyVals...)
 }
+
+func validateMasterCIDRUnchanged(old *v1alpha1.AzureConfig, new *v1alpha1.AzureConfig) error {
+	if old.Spec.Azure.VirtualNetwork.MasterSubnetCIDR != "" && old.Spec.Azure.VirtualNetwork.MasterSubnetCIDR != new.Spec.Azure.VirtualNetwork.MasterSubnetCIDR {
```
Comment (whites11) [inline pkg/azureupdate/validate_azureconfig.go:81]: allow updating the field if it was empty.
needed during CAPI cluster creation

File: pkg/azureupdate/error.go [hunk 3]
```diff
@@ -12,3 +12,12 @@ var invalidConfigError = &microerror.Error{
 func IsInvalidConfig(err error) bool {
 	return microerror.Cause(err) == invalidConfigError
 }
+
+var cantChangeMasterCIDRError = &microerror.Error{
```
Comment (tuommaki) [inline pkg/azureupdate/error.go:16]: Pure nitpick, matter of taste etc.:

"Accented" (i.e. `can't`, `won't`) language feels a bit wrong in code so could this be named a bit differently:
`masterCIDRChangeError` w/ message "disallowed":

```go
microerror.Maskf(masterCIDRChangeError, "disallowed")
```

...or more verbosely as below:
```go
microerror.Maskf(masterCIDRChangeError, "Spec.Azure.VirtualNetwork.MasterSubnetCIDR change disallowed")
```

**Review by tuommaki — APPROVED** [review]
LGTM besides comment on error naming :slightly_smiling_face:

File: pkg/azureupdate/error.go (hunk 3 above)
Comment (whites11) [reply pkg/azureupdate/error.go]: That's a good suggestion, thanks.
Renamed.

**Pull request merged**